Assignment3/WordSearch.java

```diff
@@ -0,0 +1,89 @@
+package Assignment3;
+
+import java.util.TreeSet;
+
+public class WordSearch {
+
+	/**
+	 * Find all the words from the matrix that appear in the dictionary too
+	 * 
+	 * @param rows			number of rows of the matrix
+	 * @param columns		number of columns of the matrix
+	 * @param matrix		the matrix of letters
+	 * @param dictionary		dictionary which contains a set of words
+	 * @return			the set of words which appear in the dictionary
+	 */
+	public static TreeSet<String> findWords(int rows, int columns, char[][] matrix,
+			MyDictionary dictionary) {
+		
+		/* Initialize a set in order to store the words without duplicates */
+		TreeSet<String> words = new TreeSet<>();

```

Comment:
FYI: If you did this in an interview or real life you should be able to answer why you chose a TreeSet rather than a HashSet. HashSet is normally a better choice (because of O(1) rather than O(ln N) lookups) but performance can be close. So you should be able to discuss the difference between them and make a sensible choice.